web/src/main/java/com/wilma/web/controller/web/portal/StudentPortalController.java

```diff
@@ -1,27 +1,144 @@
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.studentMenuElements,
+                "availableCategories", categoryService.findAll(),
+                "availableTags", tagService.findAll(),
+                "contentType", type,
+                "post", new PostDTO(),
+                "reply", new ReplyDTO()));
+        if(postId != null)
+            model.addAttribute("postId", postId);
+        return "/student/forum/forum-content";
+    }
+
+    @PostMapping("/reply-to-post")
+    public RedirectView replyToPost(@ModelAttribute ReplyDTO replyDTO, @RequestParam Integer postId, Model model){
+        var reply = forumService.addReplyFromDTO(replyDTO);
+        var category = reply.getPost().getCategory().getName();
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.studentMenuElements,
+                "postId", postId,
+                "categoryName", category,
+                "postsByCategory", forumService.getPostByCategoryName(category),
+                "repliesForPosts", forumService.getPostRepliesByCategory(category),
+                "reply", replyDTO));
+
+        log.info("Reply added: "+ reply);
+        return new RedirectView("/student/forum-thread?category="+ category);
+    }
+    @PostMapping("/create-post")
+    public RedirectView createPost(@ModelAttribute PostDTO postDTO, Model model){
+        var newPost = forumService.addPostFromDTO(postDTO);
+        var categoryName = newPost.getCategory().getName();
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.studentMenuElements,
+                "availableCategories", categoryService.findAll(),
+                "availableTags", tagService.findAll(),
+                "categoryName", categoryName,
+                "postsByCategory", forumService.getPostByCategoryName(categoryName),
+                "repliesForPosts", forumService.getPostRepliesByCategory(categoryName),
+                "post", postDTO));
+
+        log.info("Post created from DTO: "+ newPost);
+        return new RedirectView("/student/forum");
+    }
+
+    @GetMapping("/forum-thread")
+    public String forumThread(@RequestParam String category, Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.studentMenuElements,
+                "categoryName", category,
+                "postsByCategory", forumService.getPostByCategoryName(category),
+                "repliesForPosts", forumService.getPostRepliesByCategory(category)));
+        return "/student/forum/forum-thread";
+    }
+    //endregion
+
+    //region Resume management
+    @GetMapping("/resume-management")
     public String resumeManagement(Model model) {
         model.addAllAttributes(Map.of(
                 "currentPage", "Resume Management",
```

Comment:
Current page needs to be consistent with the element name to set the button "active"
```suggestion
"currentPage", "resume-management",
```

Reply:
Awesome, great find.